### PR TITLE
Fix rodata reservation.

### PR DESCRIPTION
**Pull request by floitsch**

### CODE REVIEW

File: components/esp_system/ld/esp32c3/sections.ld.in [hunk 1]
```diff
@@ -414,8 +414,13 @@ SECTIONS
      * driver to maintain the virtual address.
      * NOLOAD rodata may not be included in this section.
      */
+    /**
```
Comment (kasperl) [inline components/esp_system/ld/esp32c3/sections.ld.in:417]: Maybe add this comment to the c2 variant too?

**Review by kasperl — APPROVED** [review]
Looks good to me.

File: components/esp_system/ld/esp32c3/sections.ld.in (hunk 1 above)
Comment (floitsch) [reply components/esp_system/ld/esp32c3/sections.ld.in:417]: oversight.
done.

File: components/esp_system/ld/esp32c6/sections.ld.in [hunk 2]
```diff
@@ -427,8 +427,13 @@ SECTIONS
...
      * NOLOAD rodata may not be included in this section.
      */
+    /**
+     * Toit: add padding.
+     * Not completely sure this is the correct way of doing this, but we need to
+     * reserve space for the Toit containers.
+     */
     . += _esp_rodata_section_padding;
-    _rodata_reserved_end = ADDR(.flash.tbss);
+    _rodata_reserved_end = ADDR(.flash.tbss) + _eps_rodata_section_padding;
```
Comment (floitsch) [inline components/esp_system/ld/esp32c6/sections.ld.in:436]: this typo ("eps_..") was fixed in a separate commit and then force-pushed.

**Pull request merged**